drivers/flash/soc_flash_nrf.c

```diff
@@ -16,6 +16,20 @@
 #include <string.h>
 #include <nrfx_nvmc.h>
 
+#if DT_HAS_NODE(DT_INST(0, nordic_nrf51_flash_controller))
+#define DT_DRV_COMPAT nordic_nrf51_flash_controller
+#elif DT_HAS_NODE(DT_INST(0, nordic_nrf52_flash_controller))
+#define DT_DRV_COMPAT nordic_nrf52_flash_controller
+#elif DT_HAS_NODE(DT_INST(0, nordic_nrf53_flash_controller))
+#define DT_DRV_COMPAT nordic_nrf53_flash_controller
+#elif DT_HAS_NODE(DT_INST(0, nordic_nrf91_flash_controller))
+#define DT_DRV_COMPAT nordic_nrf91_flash_controller
+#else
+#error No matching compatible for soc_flash_nrf.c
+#endif
+
+#define SOC_NV_FLASH_NODE DT_INST(0, soc_nv_flash)
```

Comment:
Wouldn't it be easier to just use `DT_NODELABEL(flash_controller)` here and `DT_LABEL(SOC_NV_FLASH_NODE)` in the call to `DEVICE_AND_API_INIT()`, and avoid defining `DT_DRV_COMPAT`?

Reply:
It might, but defining DT_DRV_COMPAT at least gives some correlation between the driver and the binding(s) that it supports.  Using NODELABEL that goes away.

Reply:
Also, NODELABEL really should only be used if necessary, and in this case its not necessary.

Reply:
@anangl ?